junit-jupiter-migration-support/src/main/java/org/junit/jupiter/migrationsupport/rules/ParameterizedExtension.java

```diff
@@ -0,0 +1,292 @@
+		return extensionContext.getTestClass()
+				.flatMap(ParameterizedExtension::ensureSingleParametersMethod)
+				.filter(ReflectionUtils::isPublic);
+		// @formatter:on
+	}
+
+	private static Optional<Method> ensureSingleParametersMethod(Class<?> testClass) {
+		return ReflectionUtils.findMethods(testClass,
+			m -> m.isAnnotationPresent(Parameterized.Parameters.class)).stream().findFirst();
+	}
+
+	private static Stream<TestTemplateInvocationContext> testTemplateContextsFromParameters(Collection<Object[]> o) {
+		return o.stream().map(ParameterizedExtension::contextFactory);
+	}
+
+	private static TestTemplateInvocationContext contextFactory(Object[] parameters) {
+		return new TestTemplateInvocationContext() {
+			@Override
+			public List<Extension> getAdditionalExtensions() {
+				return singletonList(new InjectionExtension(parameters));
+			}
+		};
+	}
+
+	private static class InjectionExtension implements BeforeTestExecutionCallback {
+		private final Object[] parameters;
+
+		public InjectionExtension(Object[] parameters) {
+			this.parameters = parameters;
+		}
+
+		@Override
+		public void beforeTestExecution(TestExtensionContext context) throws Exception {
+			List<Field> parameters = parametersFields(context);
+
+			if (!parameters.isEmpty() && parameters.size() != this.parameters.length) {
+				throw unMatchedAmountOfParametersException();
+			}
+
+			for (Field param : parameters) {
+				Parameterized.Parameter annotation = param.getAnnotation(Parameterized.Parameter.class);
+				int paramIndex = annotation.value();
+				param.set(context.getTestInstance(), this.parameters[paramIndex]);
+			}
+		}
+	}
+
+	private static boolean hasArgsConstructor(ExtensionContext context) {
+		// @formatter:off
+		return context.getTestClass()
+				.map(ReflectionUtils::getDeclaredConstructor)
+				.filter(c -> c.getParameterCount() > 0)
+				.isPresent();
+		// @formatter:on
+	}
+
+	private static List<Field> parametersFields(ExtensionContext context) {
+		// @formatter:off
+		Stream<Field> fieldStream = context.getTestClass()
+				.map(Class::getDeclaredFields)
+				.map(Stream::of)
+				.orElse(Stream.empty());
+		// @formatter:on
+
+		return fieldStream.filter(f -> f.isAnnotationPresent(Parameterized.Parameter.class)).filter(
+			ReflectionUtils::isPublic).collect(toList());
+	}
+
+	private static ParameterResolutionException unMatchedAmountOfParametersException() {
+		return new ParameterResolutionException(
+			"The amount of parametersFields in the constructor doesn't match those in the provided parametersFields");
+	}
+
+	private static ParameterResolutionException wrongParametersReturnType() {
+		return new ParameterResolutionException("The @Parameters returns the wrong type");
+	}
+
+	@SuppressWarnings("unchecked")
+	private static Collection<Object[]> convertParametersMethodReturnType(Object o) {
+		if (o instanceof Collection) {
+			return (Collection<Object[]>) o;
+		}
+		else {
+			throw wrongParametersReturnType();
```

Comment:
`Parameters` methods may also return an `Iterable` or an `Object[]` (cf. https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/runners/Parameterized.java#L288-L297). You may want to use `CollectionUtils.toStream(o)` even though it supports additional types.